src/translator/byte_array_util.cpp

```diff
@@ -17,29 +18,41 @@ void * getBinaryFile(std::string path) {
         std::exit(1);
     }
     void *result;
-    int fail = posix_memalign(&result, 256, length);
-    if (fail) {
-        std::cerr << "Failed to allocate aligned memory." << std::endl;
-        std::exit(1);
+    if (isModelFile) {
+        int fail = posix_memalign(&result, 256, length);
+        if (fail) {
+            std::cerr << "Failed to allocate aligned memory." << std::endl;
+            std::exit(1);
+        }
```

Comment:
Seeing as to alignment to 256 is checked during model construction, can we shift this responsibility to the model constructor (from blob) and make this function cleaner? Maybe move as an std::string (fpath) based constructor to `MemoryGift`?

Reply:
As this alignment will be checked again during the translation, it does not hurt to remove this part. Move std::string (fpath) based constructor to MemoryGift is a bit dangerous. I was trying to keep IO apart from data.

Reply:
I'd prefer to have the validation responsibility shifted to the model constructor and make this `loadFileToMemory(fname)`. I agree on IO vs data.